### PR TITLE
MAT-6488: Implement Dialog changes for versioning. Update design syst…

### PR DESCRIPTION
**Pull request by mcmcphillips**
…em dep for dialog warning

## MADiE PR

Jira Ticket: [MAT-6488](https://jira.cms.gov/browse/MAT-6488)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [x] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/components/measureLanding/measureList/createVersionDialog/CreateVersionDialog.tsx [hunk 1]
```diff
@@ -1,94 +1,90 @@
...
+
 const CreatVersionDialog = ({
+  currentVersion,
   open,
   onClose,
   onSubmit,
   versionHelperText,
   loading,
 }) => {
+  function formikErrorHandler(name: string, isError: boolean) {
```
Comment (RohitKandimalla) [inline src/components/measureLanding/measureList/createVersionDialog/CreateVersionDialog.tsx]: isError is not required ?

File: src/components/measureLanding/measureList/createVersionDialog/CreateVersionDialog.tsx [hunk 2]
```diff
@@ -97,107 +93,130 @@ const CreatVersionDialog = ({
     },
   });
 
-  const classes = useStyles();
-  const flexEnd = classNames(classes.row, classes.end);
+  const determineVersionNumber = useCallback(() => {
```
Comment (RohitKandimalla) [inline src/components/measureLanding/measureList/createVersionDialog/CreateVersionDialog.tsx]: I thought useCallback is only helpful in stopping unnecessary renders for child components when its props didn't change. Do you think we still need it here ?
My thinking is that even if just use useEffect and call getNewVersion, unless we set a value to setNewVersionNumber, re-render doesn't happen, So having just an useEffect would be enough here ?

File: src/components/measureLanding/measureList/createVersionDialog/CreateVersionDialog.tsx (hunk 1 above)
Comment (mcmcphillips) [reply src/components/measureLanding/measureList/createVersionDialog/CreateVersionDialog.tsx]: It's not

File: src/components/measureLanding/measureList/createVersionDialog/CreateVersionDialog.tsx (hunk 2 above)
Comment (mcmcphillips) [reply src/components/measureLanding/measureList/createVersionDialog/CreateVersionDialog.tsx]: I believe so. Updated accordingly

**Review by RohitKandimalla — APPROVED** [review]

**Pull request merged**